marlowe-actus/src/Language/Marlowe/ACTUS/Model/Utility/Annuity.hs

```diff
@@ -0,0 +1,42 @@
+{-# LANGUAGE RecordWildCards #-}
+module Language.Marlowe.ACTUS.Model.Utility.Annuity where
+
+import Data.Time ( Day )
+import qualified Data.List as List ( filter, length )
+
+import Language.Marlowe.ACTUS.Definitions.ContractTerms ( DCC )
+import Language.Marlowe.ACTUS.Model.Utility.YearFraction ( yearFraction )
+
+
+calculateAnnuity ::
+  DCC -> [Day] -> Day -> Day -> Day -> Double -> Double -> Double -> Double
+calculateAnnuity dcc prSchedule maturityDate s _ n a r =
+  let scheduleTimes = List.filter (> s) prSchedule
+      m = (List.length scheduleTimes) - 1
+  in
+    (n + a) *
+    (
+      (calculateAnnuityProductFragment dcc maturityDate 0 m r scheduleTimes) /
+      (1 + (calculateAnnuitySumFragment dcc maturityDate 0 m r scheduleTimes))
```

Comment:
I would add a `let` for `(calculateAnnuitySumFragment dcc maturityDate 0 m r scheduleTimes)` or define a function for `x / (1 + x)` or `(n + a) * (x / (1 + x))`.

Reply:
This is actually quanterall implementation - but I can refactor it more :)

Reply:
Actually - I don't use this module. I can just delete it :)